### DISCUSSION TITLE
Connect K3s agent to Server over ethernet

### DISCUSSION THREAD

**Discussion by oheckmann** [k3s-io/k3s#6971] [Q&A] [answered]
I am trying to create a k3s cluster of 2 RPIs, that communicates with each other via an ethernet cable.
I installed K3s and launched the cluster using the command ` curl -sfL https://get.k3s.io | sh -s - --token token- server --cluster-init --docker --node-ip 169.254.xxx.xxx`, where the node-ip was ip address of eth0, the ethernet client of the RPI that is the server node. When the server is finished installing, I get something like this. 
```
NAME      STATUS   ROLES                       AGE   VERSION        INTERNAL-IP   
server   Ready    control-plane,etcd,master   36m   v1.25.6+k3s1   169.254.xxx.xxx
```
However, when I try to add the other node using `curl -sfL https://get.k3s.io | K3S_URL=https://<ip>.local:6443 K3S_TOKEN=toke sh -
` the system hangs forever on 'systemd: Starting k3s-agent'. An inspection of systemctl's logs shows the line `Feb 16 20:00:14 agent k3s[40021]: time="2023-02-16T20:00:14Z" level=info msg="Waiting for apiserver addresses"` constantly looping forever.

Attempting to manually join the agent to the server with the command 'sudo k3s agent --token token--server 169.254.xxx.xxx' give the error
```
INFO[0000] Starting k3s agent v1.25.6+k3s1 (9176e03c)
WARN[0000] Error starting load balancer: Initial server URL host is not defined for load balancer
FATA[0000] Initial server URL host is not defined for load balancer
```
What am I doing wrong, and what do I do to correctly connect these two nodes together. As a note, 169.254.xxx.xxx is can be pinged from the server node, so that is not the problem.

**Comment by brandond** [comment, marked as answer]
> `--node-ip 169.254.xxx.xxx`
> `K3S_URL=https://<ip>.local:6443`

You should assign real static IPs to these nodes, instead of using the Automatic Private IP Addressing (APIPA) 169.254 addresses, which are not stable and likely will change when you reboot the nodes.

You should also avoid using the mDNS `.local` hostnames, as these cannot be resolved by golang code. Configure a real DNS server, place entries in your host file, or use the bare IP addresses.

**Reply by oheckmann** [reply]
Thank you. How do you use the '--node-ip' flag when using the installation command for an agent node?

**Reply by brandond** [reply]
in exactly the same way you used it when installing the server

**Reply by oheckmann** [reply]
When I use the string 'curl -sfL https://get.k3s.io | K3S_URL=https://<server-static-ip>:6443 K3S_TOKEN=token sh - --node-ip <node-static-ip>', I get the error: "-bash: --node-ip: command not found". This is the case wherever I move it in the command

**Reply by brandond** [reply]
You're missing the `-s` flag to sh.
`curl -sfL https://get.k3s.io | sh -s - server --token=token --node-ip=x`
`curl -sfL https://get.k3s.io | sh -s - agent --token=token --node-ip=y --server=https://`

**Reply by oheckmann** [reply]
Thank you, I installed and set up both. Now I just need to figure out why ssh is breaking.
